livekit-rtc/livekit/rtc/room.py

```diff
@@ -399,9 +401,13 @@ async def disconnect(self) -> None:
         if not self.isconnected():
             return
 
+        if self._rpc_invocation_tasks:
+            for task in self._rpc_invocation_tasks:
+                task.cancel()
+            await asyncio.gather(*self._rpc_invocation_tasks, return_exceptions=True)
```

Comment:
This is only being executed when the user is explicitly calling the `disconnect` method. You should mode that at the ned of the main task

Reply:
That makes sense. I moved it to the end of `_listen_task` but I found that it no longer has the expected behavior in my example app when disconnect is called manually. So I left it in both places, I think the result will be a little messier and the caller might get the wrong error if the recipient gets forcibly disconnected but it should be ok for now.